core/pubnub_ntf_sync.c

```diff
@@ -96,12 +104,14 @@ enum pubnub_res pubnub_last_result(pubnub_t* pb)
         pbnc_fsm((pubnub_t*)pb);
     }
     result = pb->core.last_result;
+    if (result != PNR_OK){
+        pbnc_tr_cxt_state_reset(pb);
+    }
```

Comment:
In `pubnub_set_state()` call, without knowing status _PNR_OK_, I am setting state to _pb->core.state_.
In the line you pointed, when the last result is NOT successful, I am resetting the set state so that there is no orphan state in the pubnub context.